articles/realtime.md

```diff
@@ -0,0 +1,330 @@
+---
+layout: default
+title: ROS 2.0 in Realtime
+permalink: articles/realtime.html
+abstract: This article describes real-time computing requirements and how ROS 2 will fulfill them.
+published: false
+author: Jackie Kay
+---
+
+* This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+This document seeks to summarize the requirements of real-time computing and the challenges of implementing real-time performance.
+It also lays out options for how ROS 2 could be structured to enforce real-time compatibility.
+
+Robotic systems need to be responsive.
+In mission critical applications, a delay of less than a millisecond in the system can cause a catastrophic failure.
+For ROS 2.0 to capture the needs of the robotics community, the core software components must not interfere with the requirements of real-time computing.
+
+# Definition of Real-time Computing
+
+The definition of real-time computing requires the definition of a few other key terms:
+
+* Determinism: A system is deterministic if it always produces the same output for a known input. The output of a nondeterministic system will have random variations.
+* Deadline: A deadline is the finite window of time in which a certain task must be completed.
+* Quality of Service: The overall performance of a network. Includes factors such as bandwith, throughput, availability, jitter, latency, and error rates.
+
+Real-time software guarantees correct computation at the correct time.
+Hard real-time software systems have a set of strict deadlines, and missing a deadline is considered a system failure.
```

Comment:
Add a line break to separate hard real-time definition to a new paragraph. Also, consider emphasizing the hard, soft, and firm real-time initalics.